cmd/operator-sdk/generate/bundle/bundle.go

```diff
@@ -255,9 +256,39 @@ func (c bundleCmd) runMetadata() error {
 
 // generateMetadata wraps the operator-registry bundle Dockerfile/metadata generator.
 func (c bundleCmd) generateMetadata(manifestsDir, outputDir string) error {
+
+	metadataExists := checkMetatdataExists(outputDir, manifestsDir)
 	err := bundle.GenerateFunc(manifestsDir, outputDir, c.operatorName, c.channels, c.defaultChannel, c.overwrite)
 	if err != nil {
 		return fmt.Errorf("error generating bundle metadata: %v", err)
 	}
+
+	// Add SDK stamps if metadata is not present before or when overwrite is set to true.
+	if c.overwrite || !metadataExists {
+		rootDir := outputDir
+		if rootDir == "" {
+			rootDir = filepath.Dir(manifestsDir)
+		}
+
+		if err = genbundle.RewriteBundleImageContents(rootDir); err != nil {
+			return err
+		}
+	}
 	return nil
 }
+
+// checkMetatdataExists returns true if bundle.Dockerfile and metadataDir exist, if not
+// it returns false.
+func checkMetatdataExists(outputDir, manifestsDir string) bool {
+	var annotationsDir string
+	if outputDir == "" {
+		annotationsDir = filepath.Dir(manifestsDir) + bundle.MetadataDir
+	} else {
+		annotationsDir = outputDir + bundle.MetadataDir
+	}
+
+	if genutil.IsNotExist(bundle.DockerFile) || genutil.IsNotExist(annotationsDir) {
```

Comment:
To make this check robust, you need to check and write to each file individually, since one may exist without the other.

Reply:
Yes, but I was thinking of rewriting both of them even if one exists, because until tweaked manually `GenerateFunc` of OR is going to write both and also `generate bundle` provides a `--metadata` option which writes both. In the sense, if there is a scenario where only one of them exists, doesn't it mean that either it has been manually deleted there is some error (though I assume its taken care of in respective commands `create/generate`) and wouldn't rewriting **both** of them together in a clean state be a better option in such a scenario?